async-nats/src/lib.rs

```diff
@@ -698,6 +698,10 @@ impl ConnectionHandler {
         self.connection = connection;
 
         for (sid, subscription) in &self.subscriptions {
```

Comment:
We have mut here, we can filter out all closed subscriptions at this point instead without any overhead.